spec/requests/new_survey_response_spec.rb

```diff
@@ -0,0 +1,106 @@
+require 'rails_helper'
+
+describe "New Survey Response" do
+  before do
+    allow(Broadcast).to receive(:emit)
+    allow(ActionsCache).to receive(:append)
+  end
+
+  let(:action) { Action.create(form_data: {}) }
+
+  let(:params) do
+    {
+      type: 'new_survey_response',
+      params: data,
+      meta: {
+        foo: 'bar',
+        action_id: action.id
+      }
+    }
+  end
+
+  let(:data) do
+    {
+      page:         "some-page-slug",
+      name:         "Pablo José Francisco de María",
+      postal:       "W1",
+      address1:     "The Lodge",
+      address2:     "High Street",
+      city:         "London",
+      country:      "United Kingdom",
+      action_age:   "101",
+      action_foo:   "Foo",
+      action_bar:   "Bar",
+      ignored:      "ignore me",
+      email:        "omar@sumofus.org",
+      source:       'FB',
+      akid:         '3.4234.fsdf'
+    }
+  end
+
+  context "Given an action doesn't exist" do
+
+    before do
+      VCR.use_cassette("action_existing_page") do
+        post '/message', params
+      end
+    end
+
+    it 'responds successfully' do
+      expect(response).to be_success
+    end
+
+    it 'stores resource ID in champaign action' do
+      expect(action.reload.form_data['ak_resource_id']).to match(/rest\/v1\/petitionaction\//)
```

Comment:
Rubocop would complain - you should use`%r{}`, then you don't have to escape your slashes.